qlty-cli/src/upgrade.rs

```diff
@@ -182,3 +221,71 @@ struct DistManifest {
     #[serde(default)]
     announcement_tag: String,
 }
+
+fn detect_target() -> Result<&'static str> {
+    let os = std::env::consts::OS;
+    let arch = std::env::consts::ARCH;
+
+    match (os, arch) {
+        ("macos", "x86_64") => Ok("x86_64-apple-darwin"),
+        ("macos", "aarch64") => Ok("aarch64-apple-darwin"),
+        ("linux", "x86_64") => Ok("x86_64-unknown-linux-gnu"),
+        ("linux", "aarch64") => Ok("aarch64-unknown-linux-gnu"),
+        _ => bail!("Unsupported platform: {}-{}", os, arch),
```

Comment:
Windows platform unsupported. Windows users will get "Unsupported platform" error. Additionally, Windows releases use `.zip` format (not `.tar.xz`), so `extract_tarxz()` won't work. Consider documenting this limitation or adding Windows support.